### PR TITLE
Deprecate warmers

### PR DESCRIPTION
**Pull request by look**
Warmers have been deprecated as of ES 2.3 and removed from Elasticsearch as of
5.0:

https://www.elastic.co/guide/en/elasticsearch/reference/2.3/indices-warmers.html
https://www.elastic.co/guide/en/elasticsearch/reference/5.0/indices-warmers.html

I have marked Elastomer::Client::Warmer as DEPRECATED and skipped the tests when
running under ES 5.x.

### CODE REVIEW

**Comment by elireisman** [comment]
Ah crap I think I was doing the same thing (along with a couple of the other test fixes) - this LGTM!

**Comment by look** [comment]
We should consider checking the Elasticsearch version and raising errors for all methods in the `Warmer` class.

**Comment by TwP** [comment]
Checking the Elasticsearch version should be very doable but kinda tedious in places. The tedium can be dealt with though. The nice thing is we memoize the version in the `Client`, so the check will not be heavyweight at all.

https://github.com/github/elastomer-client/blob/f1a1c72c5a90c852a729b5f87f9744ea6725a695/lib/elastomer/client.rb#L54-L57

We reuse the `Client` connections, so that version will stick around.

**Comment by look** [comment]
After more discussion, @TwP and I agreed we should raise errors if you try to use the Warmer class in ES 5. The reason is that you get weird error messages now:

```
Elastomer::Client::RequestError: {"root_cause"=>[{"type"=>"invalid_type_name_exception", "reason"=>"Document mapping type name can't start with '_', found: [_warmer]"}], "type"=>"invalid_type_name_exception", "reason"=>"Document mapping type name can't start with '_', found: [_warmer]"}
```

It would be better if this failed fast.

**Comment by elireisman** [comment]
I like that idea (raising useful error) I was actually adding a SupportedVersionsError for some of the version shims for percolator API for this, but will adapt that before PR to whatever matches up here. This LGTM!

File: lib/elastomer/client/warmer.rb [hunk 1]
```diff
@@ -66,6 +72,7 @@ def get(params = {})
       # See https://github.com/elasticsearch/elasticsearch/issues/5155
       def exists?
         response = get
+
```
Comment (TwP) [inline lib/elastomer/client/warmer.rb]: Is this newline intentional?

**Comment by TwP** [comment]
Changes look good to me :rocket:

File: lib/elastomer/client/warmer.rb (hunk 1 above)
Comment (look) [reply lib/elastomer/client/warmer.rb]: I can remove it. I originally had some code here before I realized warmers were Just Gone in 5.

**Pull request merged**